homework/index.js

```diff
@@ -30,17 +34,127 @@
+    addRow(tableElement, 'Updated:', new Date(repository.updated_at).toLocaleString('en-GB'));
+  }
+
+  function setRepository(repository) {
+    const rightHand = document.getElementById('right-hand');
+    rightHand.innerHTML = '';
+    fetchJSON(repository.contributors_url)
+      .then(contributors => {
+        const h3Element = createAndAppend('h3', rightHand, {
+          text: 'Contributors:',
+        });
+        if (contributors.length === 0) {
+          h3Element.textContent = 'Contributor Not Found!';
+          return;
+        }
+        const ulElement = createAndAppend('ul', rightHand, {
+          class: 'contributors',
+        });
+        contributors.forEach(contributor => {
+          const liElement = createAndAppend('li', ulElement, {
+            class: 'contributors',
+          });
+          const anchorElement = createAndAppend('a', liElement, {
+            href: contributor.html_url,
+            target: '_blank',
+            class: 'liLink',
+          });
+          createAndAppend('img', anchorElement, {
+            class: 'user-image',
+            src: contributor.avatar_url,
+          });
+          createAndAppend('span', anchorElement, {
+            class: 'login',
+            text: contributor.login,
+          });
+          createAndAppend('span', anchorElement, {
+            class: 'counter',
+            text: contributor.contributions,
+          });
+        });
+      })
+      .catch(error => renderError(error));
+  }
+
+  function getAndAppend(url) {
+    const rootElement = document.getElementById('root');
+    const navElement = createAndAppend('nav', rootElement, {
+      class: 'nav',
+    });
+    createAndAppend('h2', navElement, {
+      class: 'heading-two',
+      text: 'HYF Repositories:',
+    });
+    const divElement = createAndAppend('div', rootElement, {
+      class: 'container',
+      id: 'container',
+    });
+    const select = createAndAppend('select', navElement);
+    fetchJSON(url)
+      .then(repositories => {
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach((elem, index) => {
+          createAndAppend('option', select, {
+            value: index,
+            text: elem.name,
+          });
+        });
+        createAndAppend('section', divElement, {
+          class: 'left-hand',
+          id: 'left-hand',
+        });
+        showRepositoryInfo(repositories[select.value]);
+        createAndAppend('section', divElement, {
+          class: 'right-hand',
+          id: 'right-hand',
+        });
+        setRepository(repositories[select.value]);
+        select.addEventListener('change', () => {
+          showRepositoryInfo(repositories[select.value]);
+          setRepository(repositories[select.value]);
+        });
+      })
+      .catch(error => renderError(error));
+  }
+
+  function main() {
+    getAndAppend(HYF_REPOS_URL);
+  }
+
+  window.onload = () => main();
```

Comment:
It is not absolutely necessary to have a function main. In your case, it just calls another function.

```js
  // function main() {
  //   getAndAppend(HYF_REPOS_URL);
  // }

  window.onload = () => getAndAppend(HYF_REPOS_URL);
}
```

Alternatively, you can directly assign your function main to `window.onload`, as long as you leave out the parentheses (otherwise you would be calling main).

```js
  function main() {
    getAndAppend(HYF_REPOS_URL);
  }

  window.onload = main;
}
```

Reply:
Hi Jim!
Thank you very much for time to review!